uptane/services/inventorydb.py

```diff
@@ -169,4 +84,122 @@ def scrub_filename(fname, expected_containing_dir):
+               ...
+             },
+    'ecu_manifests':
+             {
+               'ecu_serial': [ <full ECU manifests, in order of receipt>],
+               ...
+             }
+    'public_keys':
+             {
+               'vehicle_primaries':
+               {
+                 'vehicle_id_1': { ecu_id: 'primary_ecu_id',
+                                   public_key: <publickey for primary ecu>
+                                 },
+                 ...
+               },
+               'all_ecus':
+               {
+                 'ecu_serial_2': <publickey>,
+                 ...
+               }
+             }
+  }
+  """
+  # Inventory database dictionary
+  inventory_db = {'vehicle_manifests': {},
+                  'ecu_manifests': {},
+                  'public_keys': {
+                                  'vehicle_primaries': {},
+                                  'all_ecus': {}
+                                 }
+                 }
+
+  # Register ECU
+  def register_ecu(self, isPrimary, vin, ecu_serial, public_key):
+    if isPrimary:
+      vehicle_primary_ids = self.inventory_db['public_keys']['vehicle_primaries']
+      if (vehicle_primary_ids and (vin in vehicle_primary_ids)):
+        # rewrite value
+        (self.inventory_db['public_keys']['vehicle_primaries'])[vin] = \
+           {'ecu_id': ecu_serial, 'public_key': public_key}
+      else:
+        temp_dic = {'ecu_id': ecu_serial, 'public_key': public_key}
+        (self.inventory_db['public_keys']['vehicle_primaries'])[vin] = temp_dic
+    else:
+      ecu_serial_ids = self.inventory_db['public_keys']['all_ecus']
+      if (ecu_serial_ids and (ecu_serial in ecu_serial_ids)):
+        (self.inventory_db['public_keys']['all_ecus'])[ecu_serial] = public_key
+      else:
+        (self.inventory_db['public_keys']['all_ecus'])[ecu_serial] = public_key
+
+    is_registered = 1
+
+
+  # Save vehicle manifest
+  def save_vehicle_manifest(self, vin, signed_vehicle_manifest):
+    """
+    Given a manifest of form
+    uptane.formats.SIGNABLE_VEHICLE_VERSION_MANIFEST_SCHEMA, save it in an index
+    by vin, and save the individual ecu attestations in an index by ecu serial.
+    """
+    vehicle_primary_ids = self.inventory_db['public_keys']['vehicle_primaries']
+    ecu_serial_ids = self.inventory_db['public_keys']['all_ecus']
+    if len(vehicle_primary_ids) == 0 or len(ecu_serial_ids) == 0:
+      raise uptane.UnknownECU("InventoryDataBase has not registered yet!")
+
+    vehicle_ids = self.inventory_db['vehicle_manifests']
+    if (vehicle_ids and (vin in vehicle_ids)):
+      (self.inventory_db['vehicle_manifests'][vin]).append(signed_vehicle_manifest)
+    else:
+      vm_list = []
+      vm_list.append(signed_vehicle_manifest)
+      vehicle_ids[vin] = vm_list
+      self.inventory_db['vehicle_manifests'] = vehicle_ids
+
+
+  # Get vehicle manifest
+  def get_vehicle_manifest(self, vin):
```

Comment:
You could change the name to get_vehicle_manifest**s**() and keep returning all of them, and/or you could have get_last_vehicle_manifest() return the last one added.

In practice, I don't know if I need this get function anymore, like I said, since the values are just in a dictionary now, but it might be useful.